warehouse/migrations/versions/778f1c01a019_release_add_provides_extra_and_dynamic_.py

```diff
@@ -0,0 +1,125 @@
+Revises: 4d1b4fcc4076
+Create Date: 2023-10-12 13:04:25.039108
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+revision = "778f1c01a019"
+down_revision = "4d1b4fcc4076"
+
+
+def upgrade():
+    sa.Enum(
+        "Platform",
+        "Supported-Platform",
+        "Summary",
+        "Description",
+        "Description-Content-Type",
+        "Keywords",
+        "Home-page",
+        "Download-URL",
+        "Author",
+        "Author-email",
+        "Maintainer",
+        "Maintainer-email",
+        "License",
+        "Classifier",
+        "Requires-Dist",
+        "Requires-Python",
+        "Requires-External",
+        "Project-URL",
+        "Provides-Extra",
+        "Provides-Dist",
+        "Obsoletes-Dist",
+        name="release_dynamic_fields",
+    ).create(op.get_bind())
+    op.add_column(
+        "releases",
+        sa.Column(
+            "dynamic",
+            postgresql.ARRAY(
+                postgresql.ENUM(
+                    "Platform",
+                    "Supported-Platform",
+                    "Summary",
+                    "Description",
+                    "Description-Content-Type",
+                    "Keywords",
+                    "Home-page",
+                    "Download-URL",
+                    "Author",
+                    "Author-email",
+                    "Maintainer",
+                    "Maintainer-email",
+                    "License",
+                    "Classifier",
+                    "Requires-Dist",
+                    "Requires-Python",
+                    "Requires-External",
+                    "Project-URL",
+                    "Provides-Extra",
+                    "Provides-Dist",
+                    "Obsoletes-Dist",
+                    name="release_dynamic_fields",
+                )
+            ),
+            nullable=True,
+            comment="Array of metadata fields marked as Dynamic (PEP 643/Metadata 2.2)",
+        ),
+    )
+    op.add_column(
+        "releases",
+        sa.Column(
+            "provides_extra",
+            postgresql.ARRAY(sa.Text()),
+            nullable=True,
+            comment="Array of extra names (PEP 566/685|Metadata 2.1/2.3)",
+        ),
+    )
+
+
+def downgrade():
+    op.drop_column("releases", "provides_extra")
+    op.drop_column("releases", "dynamic")
+    sa.Enum(
+        "Platform",
+        "Supported-Platform",
+        "Summary",
+        "Description",
+        "Description-Content-Type",
+        "Keywords",
+        "Home-page",
+        "Download-URL",
+        "Author",
+        "Author-email",
+        "Maintainer",
+        "Maintainer-email",
+        "License",
+        "Classifier",
+        "Requires-Dist",
+        "Requires-Python",
+        "Requires-External",
+        "Project-URL",
+        "Provides-Extra",
+        "Provides-Dist",
+        "Obsoletes-Dist",
+        name="release_dynamic_fields",
+    ).drop(op.get_bind())
```

Comment:
This migration should probably happen first, in a separate PR. It probably should also set a higher statement timeout than the default.

Reply:
What lock/statement timeouts do you suggest?

Reply:
Adding this at the top of the migration should be sufficient:

```python
    conn = op.get_bind()
    conn.execute(sa.text("SET statement_timeout = 120000"))
```

Reply:
I've created https://github.com/pypi/warehouse/pull/15452, i unfortunately can't retarget this PR onto the other branch since the branch is in my fork